### PR TITLE
fix: hash compute again

### PR DESCRIPTION
**Pull request by lsagetlethias**
- change hash compute to be handled in main and return to renderer through IPC
- change hash compute to batch computation (1500 items per batch) => speed up compute because we don't return hash on every files
- upgrade electron
- remove direct access to electron remote object (app, dialog, shell) and use ipc instead

### CODE REVIEW

**Review by mehdilouraoui — APPROVED** [review]
💯

File: src/util/queue/queue.ts [hunk 1]
```diff
@@ -1,30 +1,33 @@
...
-            let remaining: Input[] = input;
+            let remaining: Input[][] = chunk(input, CHUNK_SIZE);
 
             const run = async () => {
                 while (remaining.length > 0) {
                     const computed = remaining[0];
                     remaining = remaining.slice(1);
-                    const value = await computingFn(computed);
+                    const values = await computingFn(computed);
+                    const newResults = values.filter((v): v is Result => isResult(v));
```
Comment (Alezco) [inline src/util/queue/queue.ts:26]:
```suggestion
                    const newResults = values.filter((value): v is Result => isResult(value));
```

File: tsconfig.json [hunk 2]
```diff
@@ -2,27 +2,36 @@
...
     "resolveJsonModule": true,
     "noUnusedLocals": true,
-    "types": ["react", "jest", "node"]       // This is a dirty-fix to ignore @types/react-native that is added
-                                     // by @types/styled-components and collides with @types/react.
-                                     // https://github.com/DefinitelyTyped/DefinitelyTyped/issues/33311#issuecomment-466731156
+    "types": [
+      "react",
+      "jest",
+      "node"
+    ] // This is a dirty-fix to ignore @types/react-native that is added
```
Comment (Alezco) [inline tsconfig.json:26]: 👍

**Review by Alezco — CHANGES_REQUESTED** [review]

File: .eslintrc.js [hunk 3]
```diff
@@ -20,7 +20,7 @@ module.exports = {
     },
   ],
   plugins: ["lodash-fp"],
-  settings: {
-    "import/resolver": "webpack",
-  },
+  // settings: {
```
[the remark this answers is not in the document]
Comment (lsagetlethias) [reply .eslintrc.js:23]: Done

File: src/common/ipc/event.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
+import { IpcMainEvent }  from "electron";
+
+/**
+ * Define an IPC config with arguments types and a return value type.
+ * 
+ * Arguments should be considerer as a spreadable array of args.
```
[the remark this answers is not in the document]
Comment (lsagetlethias) [reply src/common/ipc/event.ts:6]: Done

File: src/util/queue/queue.ts (hunk 1 above)
Comment (lsagetlethias) [reply src/util/queue/queue.ts:26]: Done

**Comment by lsagetlethias** [comment]
@Alezco Thanks pour la review, quelques oublis corrigés 😅

**Review by Alezco — APPROVED** [review]

**Pull request merged**